app/controllers/laa_references_controller.rb

```diff
@@ -80,13 +80,41 @@ def link_attempt_params
   end
 
   def resource
-    CourtDataAdaptor::Resource::LaaReference
+    if Feature.enabled?(:laa_references)
+      logger.info 'USING_V2_ENDPOINT'
+      LaaReferences
+    else
+      logger.info 'USING_V1_ENDPOINT'
+      CourtDataAdaptor::Resource::LaaReference
+    end
   end
 
   def resource_params
     @resource_params ||= @link_attempt.to_link_attributes
   end
 
+  def resource_save
+    if Feature.enabled?(:laa_references)
+      begin
+        logger.info 'CALLING_V2_MAAT_LINK'
+        @laa_reference.save!
+      rescue ActiveResource::ClientError
+        logger.info 'CLIENT_ERROR_OCCURRED'
+        render_new(I18n.t('laa_reference.link.unprocessable'), @laa_reference.errors.full_messages.join(', '))
+      rescue ActiveResource::ServerError
+        logger.error 'SERVER_ERROR_OCCURRED'
+        Sentry.capture_exception(@laa_reference.errors)
+        render_new(I18n.t('laa_reference.link.failure'), I18n.t('error.it_helpdesk'))
+      else
+        redirect_to_edit_defendants
+      end
+    else
+      logger.info 'CALLING_V1_MAAT_LINK'
+      @laa_reference.save
+      redirect_to_edit_defendants
+    end
```

Comment:
Implementation looks really good to me.

The below are suggestions  but i'm not great at OOP design in rails, so no offence taken if you prefer not to do the below or something else.

Maybe an addition could be service objects.

Like this method can call a service object or service objects for more readability, reusability, separation of concerns and ease of testing etc.
https://www.honeybadger.io/blog/refactor-ruby-rails-service-object/ 
e.g. 

```
def resource_save
   if Feature.enabled?(:laa_references)
     CDAPI::V2::LaaReference.call(resource: @laa_reference)
     # will our api just be V2? or is it YAGNI...e.g. version 3
   else
     CourtDataAdaptor::Resource::LaaReference
   end
end

in another file ------

class CDAPI::V2::LaaReference
  def self.call(laa_reference)
    new(laa_reference).call
  end
  
  def call(resource)
    laa_reference.save!
  end
  
end
```

Maybe there's a design pattern to place error handling in the `class base_model < ActiveResource` like abstract factory (i think)


Also for logging, i thought `ActiveResource::Base.logger = ActiveRecord::Base.logger
` would be enough to at least log what we're calling? If we use CDAPI::V2 in the class name...